numpy/core/arrayprint.py

```diff
@@ -607,15 +602,21 @@ def _formatArray(a, format_function, rank, max_line_len,
 
 class FloatFormat(object):
     def __init__(self, data, precision, suppress_small, sign=False):
+        # for backcompatibility, accept bools
+        if isinstance(sign, bool):
+            sign = '+' if sign else '-'
+
+        if sign == 'legacy':
+            sign = '-' if data.shape == () else ' '
+
         self.precision = precision
         self.suppress_small = suppress_small
         self.sign = sign
         self.exp_format = False
         self.large_exponent = False
-        self.max_str_len = 0
         try:
             self.fillFormat(data)
-        except (TypeError, NotImplementedError):
+        except (NotImplementedError):
```

Comment:
Parens here are weird

Reply:
Will do once we have decided whether there are other changes to make (your other comment).

By the way, the reason we can remove the TypeError is because of the addition of all the lambdas in #8963.

Reply:
Oh actually, on second look, I don't think we need the `NotImplementedError` check either, so this whole try/catch can be removed.

Reply:
I'm also going to move all of the imports of `from . import numeric as _nc` that appear at the start of 3 or 4 functions, and just make that a global import. There doesn't seem to be any recursive import problem.

Reply:
You might want to check if we have a more standard name for that than `_nc` - I think I've seen `_nx` in a bunch of places.